website/docs/rbs-support.md

```diff
@@ -63,49 +63,12 @@ an alternative annotation syntax for Ruby. The headline features of RBS:
 
 - It's the type annotation syntax blessed by the Ruby core team, and Matz in
   particular (the lead of the Ruby project).
-- RBS supports certain enticing features, like method overloading and duck
-  types.
 - The syntax is terse.
 - The Ruby distribution bundles a gem for parsing RBS annotations.
 
 However, there are a number of problems which mean that RBS syntax has a
 second-class position in Sorbet:
 
-### RBS syntax does not match the semantics of Sorbet
-
-Sorbet's type annotation syntax evolved differently from the evolution of RBS
-syntax. Unfortunately, this difference is not syntax-deep: it affects the
-semantics of the types too.
-
-Sure, there are similarities: both syntaxes offer a way to express union types,
-for example. But there are also semantic differences, and RBS syntax reflects
-these differences. Things which are possible to express in RBS syntax have no
-analogue in Sorbet and vice versa. Some examples:
-
-- RBS supports duck typing via interfaces (different from Sorbet's
-  [interfaces](abstract.md)), but
-  [Sorbet does not support duck typing](faq.md#can-i-use-sorbet-for-duck-typed-code),
-  by design.
-- Sorbet treats Ruby's `::Class` as a generic class, allowing sophisticated
```

Comment:
With RBS comments we support generic `Class`

Reply:
The point of this comment is that Sorbet supports generic `T.class_of`.

This applies to the comment below, because to my knowledge there is no way in RBS syntax to apply a type to a singleton class.

Given this:

```ruby
class RPC
  InputType = type_template
end

class MyRPC
  InputType = type_template { {fixed: Integer} }
end

sig do
  type_parameters(:U)
    .params(rpc_class: T.class_of(RPC)[RPC, T.type_parameter(:U)])
    .returns(T.type_parameter(:U))
end
def example(rpc_class) = raise

example(MyRPC)
```

How do you translate this `T.class_of` type to RBS syntax? I was under the impression that this is not a type that RBS has syntax to represent.

Reply:
Wouldn't this work?

```rbs
#: [U] (Class[RPC & U]) -> U
```

which would be the equivalent of

```rb
sig do
  type_parameters(:U)
    .params(rpc_class: T::Class[T.all(RPC, T.type_parameter(:U))])
    .returns(T.type_parameter(:U))
end
```

Or are you expressing that `example` takes a singleton class of `RPC` for which the type_template is `:U`?

Reply:
It’s the latter. The generic in my example abstracts over InputType. In your example: over `<AttachedClass>`

Reply:
I updated the change to only remove the part about generic `Class`.

btw, I don't think we have examples of the `T.class_of()[]` in the documentation?

Reply:
https://sorbet.org/docs/class-of#tclass_of-applying-type-arguments-to-a-singleton-class-type